src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -381,6 +381,155 @@ inline std::array< int, 3 > getijk( const Connection& completion ) {
     }
 }
 
+    void makeSegmentNodes(const std::size_t               last_timestep,
+                          const int                       segID,
+                          const DeckKeyword&              keyword,
+                          const std::vector<const Well*>& wells,
+                          SummaryConfig::keyword_list&    list)
+    {
+        // Modifies 'list' in place.
+        auto makeNode = [&keyword, &list]
+            (const std::string& well, const int segNumber)
+        {
+            // Grid dimensions immaterial for segment related vectors.
+            const int dims[] = { 1, 1, 1 };
+
+            list.push_back(SummaryConfig::keyword_type {
+                smspec_node_alloc(ECL_SMSPEC_SEGMENT_VAR, well.c_str(),
+                                  keyword.name().c_str(), "", ":",
+                                  dims, segNumber, 0, 0)
+            });
+        };
+
+        for (const auto* well : wells) {
+            if (! well->isMultiSegment(last_timestep)) {
```

Comment:
If we want to be totally anal this should not consider only `last_timestep` - but scan through all timesteps; if a well is MSW for an intermediate time - we should output MSW output. But what the ... Maybe they can not switch `MSW -> Normal` either?

Reply:
> If we want to be totally anal this should not consider only `last_timestep` - but scan through all timesteps;

That's a good point.  I can add that scan if you'd like.